homeassistant/components/velux/button.py

```diff
@@ -0,0 +1,51 @@
+"""Support for VELUX KLF 200 gateway button."""
+
+from __future__ import annotations
+
+from pyvlx import PyVLXException
+
+from homeassistant.components.button import ButtonDeviceClass, ButtonEntity
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import VeluxConfigEntry
+from .const import DOMAIN, LOGGER
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: VeluxConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up button entities for the Velux integration."""
+    async_add_entities([VeluxGatewayRebootButton(config_entry)])
+
+
+class VeluxGatewayRebootButton(ButtonEntity):
+    """Representation of the Velux Gateway reboot button."""
+
+    _attr_has_entity_name = True
+    _attr_translation_key = "reboot_gateway"
+    _attr_device_class = ButtonDeviceClass.RESTART
+    _attr_entity_category = EntityCategory.CONFIG
+
+    def __init__(self, config_entry: VeluxConfigEntry) -> None:
+        """Initialize the gateway reboot button."""
+        self.pyvlx = config_entry.runtime_data
```

Comment:
Personal preference would be to just pass in the client and the entry id, but leaving that up to you, now I understand your other comment in the other PR 😂